src/Core/Checkout/Document/DocumentException.php

```diff
@@ -66,4 +84,91 @@ public static function generationError(?string $message = null, ?\Throwable $e =
             $e
         );
     }
+
+    public static function customerNotLoggedIn(): self|CustomerNotLoggedInException
```

Comment:
```suggestion
    /**
     * @deprecated tag:v6.7.0 - will return only self
     */
    public static function customerNotLoggedIn(): self|CustomerNotLoggedInException
```

Since `CustomerNotLoggedInException` is marked as deprecated and will be removed, add the deprecation here for the return types too.

Reply:
Hi @Bird87ZA because I have check with `Feature::isActive('v6.7.0.0')`, so deprecate tag is not necessary

Reply:
The @deprecated tag is still helpful in that case, as developers can easily identify parts of the code that may need to be changed. For example, the IDE will often display warnings or strikethroughs of the marked method.